src/features/hotspots/checklist/HotspotChecklist.tsx

```diff
@@ -0,0 +1,267 @@
+import React, { useEffect, useState } from 'react'
+import { Hotspot } from '@helium/http'
+import { useSelector } from 'react-redux'
+import { useTranslation } from 'react-i18next'
+import { Carousel } from 'react-native-snap-carousel'
+import SkeletonPlaceholder from 'react-native-skeleton-placeholder'
+import { RootState } from '../../../store/rootReducer'
+import { useAppDispatch } from '../../../store/store'
+import { fetchChecklistActivity } from '../../../store/hotspotDetails/hotspotChecklistSlice'
+import HotspotChecklistItem from './HotspotChecklistItem'
+import { wp } from '../../../utils/layout'
+import Box from '../../../components/Box'
+import Text from '../../../components/Text'
+import CarotDown from '../../../assets/images/carot-down.svg'
+import CircleProgress from '../../../components/CircleProgress'
+import TouchableOpacityBox from '../../../components/TouchableOpacityBox'
+import animateTransition from '../../../utils/animateTransition'
+import { useSpacing } from '../../../theme/themeHooks'
+
+type Props = {
+  hotspot: Hotspot
+  witnesses?: Hotspot[]
+}
+
+type ChecklistItem = {
+  key: string
+  title: string
+  description: string
+  complete?: boolean
+  showAuto?: boolean
+  autoText?: string
+  completeText?: string
+  background?: 1 | 2 | 3 | 4
+}
+
+const HotspotChecklist = ({ hotspot, witnesses }: Props) => {
+  const dispatch = useAppDispatch()
+  const spacing = useSpacing()
+  const { t } = useTranslation()
+  const {
+    heliumData: { blockHeight },
+    hotspotChecklist: {
+      challengerTxn,
+      challengeeTxn,
+      witnessTxn,
+      dataTransferTxn,
+      loadingActivity,
+    },
+  } = useSelector((state: RootState) => state)
+
+  useEffect(() => {
+    dispatch(fetchChecklistActivity(hotspot.address))
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  const syncStatus = () => {
```

Comment:
this const and the ones below could be wrapped in `useMemo`

Reply:
Good point, updating